Sources/DataPipeline/DataPipelineImplementation.swift

```diff
@@ -76,11 +81,49 @@ class DataPipelineImplementation: DataPipelineInstance {
     var profileAttributes: [String: Any] {
         get { analytics.traits() ?? [:] }
         set {
-            let userId = analytics.userId ?? analytics.anonymousId
-            analytics.identify(userId: userId, traits: newValue)
+            let userId = analytics.userId
+            guard let userId = userId else {
+                logger.error("No user identified. If you don't have a userId but want to record traits, please pass traits using identify(body: Codable)")
+                return
+            }
+            commonIdentifyProfile(userId: userId, attributesDict: newValue)
         }
     }
 
+    private func commonIdentifyProfile(userId: String, attributesDict: [String: Any]? = nil, attributesCodable: Codable? = nil) {
```

Comment:
I don't see any checks preventing blank/empty userid in this function. 

Is that on purpose? 

I understand that maybe the segment SDK accepts blank/empty identifiers but maybe that will put our SDK into an invalid state as it has in the past?

Reply:
I am not sure what PR it was, but @Shahroz16, @ami-aman and I had this conversation. And I believe we decided it was best to prevent blank/empty identifiers by logging a warning and ignoring the identify request.